taskoutput/task_output.go

```diff
@@ -1,32 +1,33 @@
 package taskoutput
 
-import (
-	"github.com/evergreen-ci/evergreen/taskoutput/tasklogs"
-	"github.com/evergreen-ci/evergreen/taskoutput/testlogs"
-)
+import "github.com/evergreen-ci/evergreen"
 
 // TaskOutput is the versioned entry point for coordinating persistent storage
-// of a task run's output data. The int value represents the centralized
-// version of the task output, which then maps to the underlying versions of
-// each recognized output type collected during a task run.
-type TaskOutput int
+// of a task run's output data.
+type TaskOutput struct {
+	TaskLogs TaskLogOutput `bson:"task_logs,omitempty" json:"task_logs"`
+	TestLogs TestLogOutput `bson:"test_logs,omitempty" json:"test_logs"`
+}
 
-// TaskLogs returns the versioned entry point of the task logs output type.
-func (o TaskOutput) TaskLogs() tasklogs.TaskLogs {
-	switch {
-	case o >= 0:
-		return tasklogs.TaskLogs(0)
-	default:
-		return tasklogs.TaskLogs(-1)
-	}
+// TaskOptions represents the task-level information required for accessing
+// task logs belonging to a task run.
+type TaskOptions struct {
```

Comment:
nit: would be worth adding `bson:"-" json:"-"` to this (+ `TaskLogGetOptions`) to make it clear that it's not meant to be serialized (or otherwise, add tags names).